### PR TITLE
Bug 1784201: *: support day-2 proxy setup

### PR DESCRIPTION
**Pull request by runcom**
Signed-off-by: Antonio Murdaca <runcom@linux.com>

### CODE REVIEW

File: cmd/common/helpers.go [hunk 1]
```diff
@@ -49,3 +57,67 @@ func CreateResourceLock(cb *clients.Builder, componentNamespace, componentName s
...
+				glog.Errorf("failed to read watched file %q: %v", defaultTrustedCABundle, err)
+				close(stopCh)
+				return
+			}
+			if !bytes.Equal(trustedCAWatcher.original, latest) {
+				glog.Infof("watched file %q changed, stopping operator", defaultTrustedCABundle)
+				close(stopCh)
+				return
+			}
+		case err := <-trustedCAWatcher.watcher.Errors:
```
Comment (ashcrow) [inline cmd/common/helpers.go:117]: Just to verify, if the watcher errors we expect to continue watching?

Comment (runcom) [reply cmd/common/helpers.go:117]: Yep, can be transient and if it’s not we’ll see that in logs

**Comment by runcom** [comment]
I should hold this tho, it’s still under testing with real a real proxy, I’ll unblock once I’m done testing

/hold

**Comment by runcom** [comment]
correct fix is here https://github.com/openshift/machine-config-operator/pull/1714

**Pull request closed without merging**